### PR TITLE
[build]|[tensorflow] Add Dockerfiles for tf-2.3 training

### PR DESCRIPTION
**Pull request by satishpasumarthi**
*Issue #, if available:*

## Checklist
- [x] I've prepended PR tag with frameworks/job this applies to : [mxnet, tensorflow, pytorch] | [build] | [test] | [build, test] | [ec2, ecs, eks, sagemaker]
- [x] (If applicable) I've documented below the DLC image/dockerfile this relates to
- [x] (If applicable) I've documented below the tests I've run on the DLC image
- [x] (If applicable) I've reviewed the licenses of updated and new binaries and their dependencies to make sure all licenses are on the Apache Software Foundation Third Party License Policy Category A or Category B license list.  See [https://www.apache.org/legal/resolved.html](https://www.apache.org/legal/resolved.html).
- [x] (If applicable) I've scanned the updated and new binaries to make sure they do not have vulnerabilities associated with them.

*Description:*
Docker files for TF 2.3 release
package changed:
    * Upgrade libnccl2 and libnccl2-dev version 
        * 2.6.4 → 2.7.6
    * Upgrade libcublas10 and libcublas10-dev
        * 10.2.1.243-1 → 10.2.2.89-1 
    * Upgrade openmpi
        * 4.0.1 -> 4.0.4

FYI:
Latest releases are available for some of the python packages: (we didn’t change them in the docker file.)
numpy - 1.19.1
Spicy - 1.5.2
pandas - 1.1
pillow - 7.2.0
opencv-python 4.3.0.36
requests - 2.24.0
psutil - 5.7.2

We have tested the TF binaries using TF addons version v0.10.0 (latest)
*Tests run:*
- PR tests

*DLC image/dockerfile:*
tensorflow/training/docker/2.3.0/py3/Dockerfile.cpu
tensorflow/training/docker/2.3.0/py3/cu102/Dockerfile.gpu

*Additional context:*

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license. I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: tensorflow/training/docker/2.3.0/py3/Dockerfile.cpu [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+    Pillow==7.1.2 \
+    h5py==2.10.0 \
+    python-dateutil==2.8.1 \
+    pyYAML==5.3.1 \
+    requests==2.23.0 \
+    "awscli<2" \
+    mpi4py==3.0.3 \
+    opencv-python==4.2.0.32 \
+    "sagemaker<2" \
+    sagemaker-experiments==0.1.13 \
```
Comment (TusharKanekiDey) [inline tensorflow/training/docker/2.3.0/py3/Dockerfile.cpu:153]: @Danabens, Can we keep it as sagemaker-experiments==0.* \ to fetch the latest version like PT.

Comment (danabens) [reply tensorflow/training/docker/2.3.0/py3/Dockerfile.cpu:153]: ya go ahead and unpin it

File: tensorflow/training/docker/2.3.0/py3/Dockerfile.cpu [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+    "sagemaker-tensorflow>=2.2,<2.3" \
```
Comment (TusharKanekiDey) [inline tensorflow/training/docker/2.3.0/py3/Dockerfile.cpu:154]: @ajaykarpur, Can you please check the sagemaker related package are the latest ones ?

**Comment by TusharKanekiDey** [comment]
@satishpasumarthi Please change the buildspec of the tensorflow (https://github.com/aws/deep-learning-containers/blob/master/tensorflow/buildspec.yml) so that it builds and test the required Images.

**Comment by arjkesh** [comment]
Hi @satishpasumarthi -- can you update the PR to include changes to buildspec.yml, specifying which images you want to build/test? Additionally, are there any test changes/additions needed for the tf 2.3 release?

**Comment by arjkesh** [comment]
with respect to these packages:
```
numpy - 1.19.1
Spicy - 1.5.2
pandas - 1.1
pillow - 7.2.0
opencv-python 4.3.0.36
requests - 2.24.0
psutil - 5.7.2
```

Can we unpin some/all of these? The only one I'd be concerned about is opencv-python, is that OK to unpin as well?

**Comment by satishpasumarthi** [comment]
> test changes/additions needed for the tf 2.3 release?

updated buildspec.yml file. There are no additional tests required for 2.3. https://github.com/aws/deep-learning-containers/pull/424/files is still WIP and not a blocker for the release. 
@sboshin

**Comment by sboshin** [comment]
The decision to update those packages, pinning, and unpinning are up to you @arjkesh and the team. This was just added because I remembered it as part of the old process.

File: tensorflow/training/docker/2.3.0/py3/Dockerfile.cpu (hunk 2 above)
Comment (sboshin) [reply tensorflow/training/docker/2.3.0/py3/Dockerfile.cpu:154]: This PR should not be gated by Sagemaker packages.

**Comment by arjkesh** [comment]
I think we need to update smdebug to 0.8.2 -- the smdebug tests are currently failing. Could you increase this version?

**Comment by arjkesh** [comment]
smdebug test is currently failing, looks like pipemode sagemaker tests are failing as well

**Comment by sboshin** [comment]
> smdebug test is currently failing, looks like pipemode sagemaker tests are failing as well
As pointed out, sagemaker-tensorflow isn't updated. Can we have the relevant teams update the packages as needed?

**Comment by nskool** [comment]
> with respect to these packages:
> 
> ```
> numpy - 1.19.1
> Spicy - 1.5.2
> pandas - 1.1
> pillow - 7.2.0
> opencv-python 4.3.0.36
> requests - 2.24.0
> psutil - 5.7.2
> ```
> 
> Can we unpin some/all of these? The only one I'd be concerned about is opencv-python, is that OK to unpin as well?

For purposes of repeatability, it's better to pin these to the latest available packages at the time of PR creation. Since these are some of the more critical packages, could you please update them to the latest?

**Comment by satishpasumarthi** [comment]
> > with respect to these packages:
> > ```
> > numpy - 1.19.1
> > Spicy - 1.5.2
> > pandas - 1.1
> > pillow - 7.2.0
> > opencv-python 4.3.0.36
> > requests - 2.24.0
> > psutil - 5.7.2
> > ```
> > 
> > 
> > Can we unpin some/all of these? The only one I'd be concerned about is opencv-python, is that OK to unpin as well?
> 
> For purposes of repeatability, it's better to pin these to the latest available packages at the time of PR creation. Since these are some of the more critical packages, could you please update them to the latest?

Updated the Dockerfiles.

**Comment by arjkesh** [comment]
Temporarily added the updates from #442 so we don't have to wait for that PR merge in order to get feedback on what's wrong with the SMDebug test.

**Comment by arjkesh** [comment]
I notice we install sagemaker pysdk inside these containers. Do we have any tests that actually launch a sagemaker job from within a container?

File: tensorflow/training/docker/2.3.0/py3/cu102/Dockerfile.gpu [hunk 3]
```diff
@@ -0,0 +1,224 @@
...
+    h5py==2.10.0 \
+    python-dateutil==2.8.1 \
+    pyYAML==5.3.1 \
+    requests==2.24.0 \
+    "awscli<2" \
+    mpi4py==3.0.3 \
+    opencv-python==4.3.0.36 \
+    "sagemaker<2" \
+    sagemaker-experiments==0.* \
+    "sagemaker-tensorflow>=2.2,<2.3" \
```
Comment (anankira) [inline tensorflow/training/docker/2.3.0/py3/cu102/Dockerfile.gpu:198]: @satishpasumarthi please update sagemaker-tensorflow* to 2.3 version

https://pypi.org/project/sagemaker-tensorflow/
https://pypi.org/project/sagemaker-tensorflow-training/

Comment (arjkesh) [reply tensorflow/training/docker/2.3.0/py3/cu102/Dockerfile.gpu]: @satishpasumarthi just a heads up, I made this change

**Pull request merged**